core/sys/linux/file.odin

```diff
@@ -0,0 +1,1697 @@
+FCntl_Command_SET_RW_HINT      :: distinct FCntl_Command
+FCntl_Command_GET_FILE_RW_HINT :: distinct FCntl_Command
+FCntl_Command_SET_FILE_RW_HINT :: distinct FCntl_Command
+F_DUPFD            :: FCntl_Command_DUPFD(.DUPFD)
+F_GETFD            :: FCntl_Command_GETFD(.GETFD)
+F_SETFD            :: FCntl_Command_SETFD(.SETFD)
+F_GETFL            :: FCntl_Command_GETFL(.GETFL)
+F_SETFL            :: FCntl_Command_SETFL(.SETFL)
+// F_GETLK64          :: FCntl_Command_GETLK64(.GETLK64)
+// F_SETLK64          :: FCntl_Command_SETLK64(.SETLK64)
+// F_SETLKW64         :: FCntl_Command_SETLKW64(.SETLKW64)
+F_GETLK            :: FCntl_Command_GETLK(.GETLK)
+F_SETLK            :: FCntl_Command_SETLK(.SETLK)
+F_SETLKW           :: FCntl_Command_SETLKW(.SETLKW)
+F_DUPFD_CLOEXEC    :: FCntl_Command_DUPFD_CLOEXEC(.DUPFD_CLOEXEC)
+F_SETOWN           :: FCntl_Command_SETOWN(.SETOWN)
+F_GETOWN           :: FCntl_Command_GETOWN(.GETOWN)
+F_SETSIG           :: FCntl_Command_SETSIG(.SETSIG)
+F_GETSIG           :: FCntl_Command_GETSIG(.GETSIG)
+F_SETOWN_EX        :: FCntl_Command_SETOWN_EX(.SETOWN_EX)
+F_GETOWN_EX        :: FCntl_Command_GETOWN_EX(.GETOWN_EX)
+F_SETLEASE         :: FCntl_Command_SETLEASE(.SETLEASE)
+F_GETLEASE         :: FCntl_Command_GETLEASE(.GETLEASE)
+F_NOTIFY           :: FCntl_Command_NOTIFY(.NOTIFY)
+F_SETPIPE_SZ       :: FCntl_Command_SETPIPE_SZ(.SETPIPE_SZ)
+F_GETPIPE_SZ       :: FCntl_Command_GETPIPE_SZ(.GETPIPE_SZ)
+F_ADD_SEALS        :: FCntl_Command_ADD_SEALS(.ADD_SEALS)
+F_GET_SEALS        :: FCntl_Command_GET_SEALS(.GET_SEALS)
+F_GET_RW_HINT      :: FCntl_Command_GET_RW_HINT(.GET_RW_HINT)
+F_SET_RW_HINT      :: FCntl_Command_SET_RW_HINT(.SET_RW_HINT)
+F_GET_FILE_RW_HINT :: FCntl_Command_GET_FILE_RW_HINT(.GET_FILE_RW_HINT)
+F_SET_FILE_RW_HINT :: FCntl_Command_SET_FILE_RW_HINT(.SET_FILE_RW_HINT)
+
+
+/// Get file status
+///
+/// Returns information about the file in struct pointed to by `stat` parameter.
+///
+/// Available since Linux 1.0
+/// For 32-bit systems a different syscall is used that became available since 2.4
+/// Not available on arm64
+stat :: proc "contextless" (filename: cstring, stat: ^Stat)->(Errno) {
+	when LONG_BITS == 64 {
+		when ARCH == "arm64" {
+			ret := syscall(SYS_fstatat, AT_FDCWD, cast(rawptr) filename, stat)
+			return Errno(-ret)
+		} else {
+			ret := syscall(SYS_stat, cast(rawptr) filename, stat)
+			return Errno(-ret)
+		}
+	} else {
+		ret := syscall(SYS_stat64, cast(rawptr) filename, stat)
+		return Errno(-ret)
+	}
+}
+
+/// Get file status from file descriptor
+///
+/// Returns information about the file in struct pointed to by `stat` parameter.
+///
+/// Available since Linux 1.0
+/// For 32-bit systems a different syscall is used that became available since 2.4
+fstat :: proc "contextless" (fd: Fd, stat: ^Stat)->(Errno) {
+	when LONG_BITS == 64 {
+		ret := syscall(SYS_fstat, stat)
+		return Errno(-ret)
+	} else when LONG_BITS == 32 {
+		ret := syscall(SYS_fstat64, stat)
+		return Errno(-ret)
```

Comment:
```odin
ret := syscall(SYS_fstat when LONG_BITS == 64 else SYS_fstat64, stat)
return Errno(-ret)
```

Reply:
Might have to refuse this suggestion... I don't like long lines, and I feel like its much clearer to pattern-match the difference between two blocks than read an inline when statement

Reply:
That's fine. It was only a suggestion.